extensions/emmet/src/abbreviationActions.ts

```diff
@@ -616,14 +616,30 @@ function expandAbbr(input: ExpandAbbreviationInput): string | undefined {
 				}
 
 				// If wrapping with a block element, insert newline in the text to wrap.
-				if (wrappingNode && inlineElements.indexOf(wrappingNode.name) === -1) {
-					wrappingNode.value = '\n\t' + wrappingNode.value + '\n';
+				if (wrappingNode) {
+					if (inlineElements.indexOf(wrappingNode.name) === -1) {
+						wrappingNode.value = '\n\t' + wrappingNode.value + '\n';
+					}
+
+					// Fixes #54711
+					if (wrappingNode.name === 'a') {
+						let hrefAtt = wrappingNode.attributes.filter((a) => a.name === 'href')[0];
+						if (hrefAtt) {
+							hrefAtt.value = input.textToWrap[0]; // This is safe because we've already checked that length is 1.
+						} else {
+							wrappingNode.attributes.push({
```

Comment:
What case are we trying to cover here? If it is an `a` tag then emmet would have added the `href` attribute for sure even if it has empty value

Reply:
Yes, lets drop that.